modules/gapi/src/backends/ie/giebackend/giewrapper.cpp

```diff
@@ -93,11 +95,34 @@ IE::InferencePlugin giewrap::getPlugin(const GIEParam& params) {
     return plugin;
 }
 #else // >= 2019.R2
-IE::Core giewrap::getCore() {
+
+static IE::Core create_IE_Core_pointer() {
+    // NB: 'delete' is never called
+    static IE::Core* core = new IE::Core();
+    return *core;
+}
+
+static IE::Core create_IE_Core_instance() {
     static IE::Core core;
     return core;
 }
 
+IE::Core giewrap::getCore() {
+    // NB: to make happy memory leak tools use:
+    // - OPENCV_GAPI_INFERENCE_ENGINE_CORE_LIFETIME_WORKAROUND=0
+    static bool param_GAPI_INFERENCE_ENGINE_CORE_LIFETIME_WORKAROUND =
+        utils::getConfigurationParameterBool(
+                "OPENCV_GAPI_INFERENCE_ENGINE_CORE_LIFETIME_WORKAROUND",
```

Comment:
please put it on Wiki. Also, shouldn't we reuse the same name as in DNN module?

Reply:
>>> shouldn't we reuse the same name as in DNN module?
It seems to be confusing of using `DNN` in flag name, moreover this logic is a bit different than it's done in `dnn` module, since some of the hacks aren't needed anymore.

Reply:
ok